src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,105 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handlerimpl.BalanceOperation;
+import core.basesyntax.strategy.handlerimpl.PurchaseOperation;
+import core.basesyntax.strategy.handlerimpl.ReturnOperation;
+import core.basesyntax.strategy.handlerimpl.SupplyOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class FruitShopServiceImplTest {
+    public static final Integer BANANA_AMOUNT = 152;
+    public static final Integer APPLE_AMOUNT = 90;
+    private static FruitShopService fruitShopService;
+    private static List<FruitTransaction> fruitTransactions;
+    private static Map<String,Integer> expectedFruitStorage;
+
+    @BeforeClass
+    public static void setUp() {
```

Comment:
It's great, but to be honest, I don't see the importance of this initialization test. If it works correctly, then in the writer test you check the correctness of the result, in the balance - BALANCE. And only on null (incorrectData_notOk) is kind of relevant.